### PR TITLE
Sidebar changes broke the modules xtd button

### PR DESCRIPTION
**Pull request by dgrammatiko**
#### The Problem

Login in front end and try to edit an article
use the XTD button modules to insert a module
oops error 500
#### The fix

The changes for the sidebar broke it so lets be more specific on those lines and exempt them from the modal
#### Testing

Apply patch
Repeat the above procedure, the modal gets the correct content

### CODE REVIEW

**Comment by zero-24** [comment]
I have tested this item :white_check_mark: successfully on eb142c8087a7ea96637403b6bc5090e9ba4021ac<br /><br />Works. To get the insert finaly work you need also apply this: https://github.com/joomla/joomla-cms/pull/8269<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8272">issues.joomla.org/joomla-cms/8272</a>.</sub>

**Comment by roland-d** [comment]
I have tested this item :white_check_mark: successfully on eb142c8087a7ea96637403b6bc5090e9ba4021ac<br /><br />The modal gives an error when trying to open it in a frontend article. After applying the fix, the modal lets me choose a module and inserts it into the text.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8272">issues.joomla.org/joomla-cms/8272</a>.</sub>

**Comment by zero-24** [comment]
@joomla-cms-bot the label please :smiley: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8272">issues.joomla.org/joomla-cms/8272</a>.</sub>

File: administrator/components/com_modules/views/modules/view.html.php [hunk 1]
```diff
@@ -34,8 +34,11 @@ public function display($tpl = null)
 		$this->items		= $this->get('Items');
 		$this->pagination	= $this->get('Pagination');
 		$this->state		= $this->get('State');
-		$this->filterForm       = $this->get('FilterForm');
-		$this->activeFilters    = $this->get('ActiveFilters');
+		if ($this->getLayout() == 'default')
```
Comment (roland-d) [inline administrator/components/com_modules/views/modules/view.html.php]: @dgt41 We need an empty line before the if statement. Thanks.

File: administrator/components/com_modules/views/modules/view.html.php [hunk 2]
```diff
@@ -34,8 +34,11 @@ public function display($tpl = null)
...
+		{
+			$this->filterForm       = $this->get('FilterForm');
```
Comment (zero-24) [inline administrator/components/com_modules/views/modules/view.html.php]: can we remove the spaces that are too much now ;)

**Comment by zero-24** [comment]
After testing an CS. Good to merge. Thanks.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8272">issues.joomla.org/joomla-cms/8272</a>.</sub>

**Comment by Kubik-Rubik** [comment]
Thank you @dgt41 for the fast fix! Merged.

**Pull request merged**